pxtcompiler/emitter/service.ts

```diff
@@ -1897,6 +1897,22 @@ namespace ts.pxtc.service {
 
             // check for explicit default in the attributes
             if (attrs && attrs.paramDefl && attrs.paramDefl[name]) {
+                if (typeNode.kind == SK.AnyKeyword) {
+                    const defaultName = attrs.paramDefl[name].toUpperCase();
+                    if (!Number.isNaN(+defaultName)) {
+                        // try to parse as a number
+                        typeNode.kind = SK.NumberKeyword;
+                    } else if (defaultName == "FALSE" || defaultName == "TRUE") {
```

Comment:
should probably lowercase it before checking, in TS it's lowercase and in python it's uppercase

Reply:
never mind! I can't read

Reply:
ahaha i can see into the future